ext/crates/fp/src/matrix/subspace.rs

```diff
@@ -209,18 +214,34 @@ impl Subspace {
             .map(FpVector::as_slice)
             .take(self.dimension())
     }
+
+    pub fn sum(&self, other: &Self) -> Self {
+        let self_rows = self.matrix.clone().into_iter();
+        let other_rows = other.matrix.clone().into_iter();
+        let new_rows = self_rows.chain(other_rows).collect();
+
+        let mut ret = Self::from_matrix(Matrix::from_rows(self.prime(), new_rows));
+        ret.matrix.trim(0, self.matrix.columns() + 1, 0);
+        ret
+    }
 }
 
 impl std::fmt::Display for Subspace {
     fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
         let dim = self.dimension();
-        for row in self.matrix.iter().take(dim) {
-            if f.alternate() {
-                writeln!(f, "{row:#}")?;
-            } else {
-                writeln!(f, "{row}")?;
-            }
-        }
+        let mut rows = self.matrix.iter().take(dim);
+
+        let output = if f.alternate() {
```

Comment:
What does `f.alternate()` mean?

Reply:
It formats the subspace in one line, to anticipate printing Massey products to `stdout`. I'll see if I can add a docstring.

Reply:
Is there some choice of the name? What about calling it `one_line` or something a bit more explicit?

Reply:
It's a method on `std::fmt::Formatter` so we're stuck with that. In general for anything that implements Display you can print it normally using "{foo}" or use an alternate format with "{foo:#}".

Reply:
I added a doctest. Since it's part of the `Subspace` API it's also included in #148